### PR TITLE
Separate filter form and body form in table

### PR DESCRIPTION
**Pull request by goemen**
- Fix financial value filters and adding tooltips

### CODE REVIEW

File: frontend/src/components/Table/Table.tsx [hunk 1]
```diff
@@ -498,7 +512,7 @@ const Table = <T extends object>(props: PropsWithChildren<TableProps<T>>): React
                 innerRef={filterFormRef as any}
               >
                 {actions => (
-                  <>
+                  <Form>
```
Comment (agalbrai) [inline frontend/src/components/Table/Table.tsx:515]: How does this change the component? Does it add behaviour of some kind?

File: frontend/src/features/properties/list/PropertyListView.tsx [hunk 2]
```diff
@@ -490,113 +501,135 @@ const PropertyListView: React.FC = () => {
...
+                const changedRows = dirtyRows.map(change => {
+                  const data = { ...values.properties![change.rowId] };
+
+                  return { rowId: change.rowId, data } as any;
+                });
+                if (changedRows.length > 0) {
+                  for (const change of changedRows) {
+                    const apiProperty = toApiProperty(change.data as any, true);
+                    const callApi = apiProperty.parcelId ? updateParcel : updateBuilding;
+                    const response: any = await callApi(apiProperty.id, apiProperty);
```
Comment (devinleighsmith) [inline frontend/src/features/properties/list/PropertyListView.tsx:608]: does this perform ok if the user changes a relatively large number of rows at the same time? What happens if one of the requests fails?

File: frontend/src/features/properties/list/PropertyListView.tsx [hunk 3]
```diff
@@ -490,113 +501,135 @@ const PropertyListView: React.FC = () => {
...
+          }}
+          renderBodyComponent={({ body }) => (
+            <Formik
+              innerRef={tableFormRef as any}
+              initialValues={{ properties: data || [] }}
+              enableReinitialize
+              onSubmit={async values => {
+                let nextProperties = [...values.properties];
+                const changedRows = dirtyRows.map(change => {
+                  const data = { ...values.properties![change.rowId] };
```
Comment (devinleighsmith) [inline frontend/src/features/properties/list/PropertyListView.tsx:600]: this works properly with paging, right?

File: frontend/src/features/properties/list/PropertyListView.tsx [hunk 4]
```diff
@@ -490,113 +501,135 @@ const PropertyListView: React.FC = () => {
...
+            onExpand: loadBuildings,
+            getRowId: row => row.id,
+          }}
+          filter={appliedFilter}
+          onFilterChange={values => {
+            setFilter({ ...filter, ...values });
+          }}
+          renderBodyComponent={({ body }) => (
+            <Formik
+              innerRef={tableFormRef as any}
```
Comment (devinleighsmith) [inline frontend/src/features/properties/list/PropertyListView.tsx:594]: to we not have any validation on this form? note that while chrome doesn't allow you to type non-numeric characters into numeric fields other browsers will.

File: frontend/src/features/properties/list/PropertyListView.tsx [hunk 5]
```diff
@@ -490,113 +501,135 @@ const PropertyListView: React.FC = () => {
...
-                } else {
-                  return false;
-                }
-              }}
-              detailsPanel={{
-                render: val => {
-                  if (expandData[val.id]) {
-                    return <Buildings hideHeaders={true} data={expandData[val.id]} />;
+          canRowExpand={(val: any) => {
+            if (val.values.propertyTypeId === 0) {
```
Comment (devinleighsmith) [inline frontend/src/features/properties/list/PropertyListView.tsx:568]: This probably needs to include subdivisions now.

File: frontend/src/features/properties/list/PropertyListView.tsx [hunk 6]
```diff
@@ -490,113 +501,135 @@ const PropertyListView: React.FC = () => {
...
+          }}
+          filter={appliedFilter}
+          onFilterChange={values => {
+            setFilter({ ...filter, ...values });
+          }}
+          renderBodyComponent={({ body }) => (
+            <Formik
+              innerRef={tableFormRef as any}
+              initialValues={{ properties: data || [] }}
+              enableReinitialize
```
Comment (devinleighsmith) [inline frontend/src/features/properties/list/PropertyListView.tsx:596]: probably will want to make sure that leaving the enableReinitialize isn't refreshing the form if a user is in edit mode for  a while ( a few minutes) also want to make sure that changes in the filter bar affect edit mode in a reasonable way.

**Review by devinleighsmith — APPROVED** [review]
Approving, but I left some comments on areas where I'm concerned about possible issues.

File: frontend/src/features/properties/list/PropertyListView.tsx (hunk 5 above)
Comment (Fosol) [reply frontend/src/features/properties/list/PropertyListView.tsx:568]: Will subdivisions have buildings on them?

Comment (Fosol) [reply frontend/src/features/properties/list/PropertyListView.tsx:568]: Don't we have an `enum` for `PropertyTypes`?

File: frontend/src/features/properties/list/PropertyListView.tsx (hunk 2 above)
Comment (Fosol) [reply frontend/src/features/properties/list/PropertyListView.tsx:608]: You need to handle if one or more of the requests fail

**Review by Fosol — APPROVED** [review]

File: frontend/src/components/Table/Table.tsx (hunk 1 above)
Comment (goemen) [reply frontend/src/components/Table/Table.tsx:515]: This allows to submit the form by clicking ENTER

File: frontend/src/features/properties/list/PropertyListView.tsx (hunk 3 above)
Comment (goemen) [reply frontend/src/features/properties/list/PropertyListView.tsx:600]: Only the current page is part of the form, yes is works with paging

File: frontend/src/features/properties/list/PropertyListView.tsx (hunk 2 above)
Comment (goemen) [reply frontend/src/features/properties/list/PropertyListView.tsx:608]: Added a toast display and added row fields validation error display

File: frontend/src/features/properties/list/PropertyListView.tsx (hunk 4 above)
Comment (goemen) [reply frontend/src/features/properties/list/PropertyListView.tsx:594]: Added validation

File: frontend/src/features/properties/list/PropertyListView.tsx (hunk 5 above)
Comment (goemen) [reply frontend/src/features/properties/list/PropertyListView.tsx:568]: @devinleighsmith can you verify if this change need to be made, not too familiar with subdivisions in the table?

File: frontend/src/features/properties/list/PropertyListView.tsx (hunk 6 above)
Comment (goemen) [reply frontend/src/features/properties/list/PropertyListView.tsx:596]: This seems to be not a problem

**Pull request merged**